cmd/kubelet/app/server.go

```diff
@@ -513,6 +537,47 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func initializeCertificateManager(certDirectory string, certRotation int32, nodeName types.NodeName, certData []byte, keyData []byte) (certificate.Manager, error) {
+	certificateStore, err := certificate.NewFileStore(
+		"kubelet-client",
+		certDirectory,
+		certDirectory,
+		"",
+		"")
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize certificate store: %v", err)
+	}
+	var rotation uint
+	if certRotation > 0 {
+		rotation = uint(certRotation)
+	}
+	clientCertificateManager, err := certificate.NewManager(&certificate.Config{
+		Template: &x509.CertificateRequest{
+			Subject: pkix.Name{
+				Organization: []string{"system:nodes"},
+				CommonName:   fmt.Sprintf("system:node:%s", nodeName),
+			},
+		},
+		Usages: []certificates.KeyUsage{
+			certificates.UsageDigitalSignature,
+			certificates.UsageKeyEncipherment,
+			certificates.UsageClientAuth,
+		},
+		CertificateStore:           certificateStore,
+		CertificateRotationPercent: rotation,
+		Rotation: func() {
+			glog.Info("The client certificate was rotated, restarting so changes take effect.")
+			os.Exit(0)
```

Comment:
Client GetCertificate is not supported until go1.8. Any suggestions pretaining to how to do this. I'd like to point out that this is an experimantal flag and rotations event frequency is very low (O(months)).